core/modules/layout/layout.admin.inc

```diff
@@ -885,6 +899,78 @@ function layout_settings_form_validate(&$form, &$form_state) {
       form_error($form['conditions']['active'][$key], t('The condition \'!condition\' does not have the required contexts at this path. Remove the condition to save the layout.', array('!condition' => $condition->summary())));
     }
   }
+
+  global $_layout_additional_paths;
```

Comment:
Why we need a global variable here? Shouldnt all settings values be saved in `$layout->settings`?

Reply:
Agreed, this would be a lot better as a property on the object. It could be managed in the same way as the `$layout->removed_blocks` property, where it's used to hold values between different operations. We could introduce `$layout->added_paths` or similar to hold this between validate and save operations.